src/client/systems/NetworkingClient.cpp

```diff
@@ -263,14 +259,33 @@ void NetworkingClient::unreliableRecv() {
         if (status != sf::Socket::NotReady) {
             GameState game_state;
             packet >> game_state;
-            std::lock_guard<std::mutex> m_gameState_guard(m_gameState_lock);
-            m_gameState_t = game_state;
+            updateGameStateBuffer(game_state);
         }
 
         std::this_thread::sleep_for(GAME_SETTINGS.CLIENT_UNRELIABLE_RECV_SLEEP);
     }
 }
 
+void NetworkingClient::updateGameStateBuffer(GameState& game_state) {
+    std::lock_guard<std::mutex> m_gameStateBuffer_guard(m_gameStateBuffer_lock);
+    std::lock_guard<std::mutex> m_serverGameStateMetric_guard(m_serverGameStateMetric_lock);
+    m_gameStateBuffer_t[game_state.core.tick] = game_state;
```

Comment:
im guessing a buffer overflow is impossible here since I believe this is a circular buffer iirc

Reply:
it's a map not a circular buffer